### DISCUSSION TITLE
[4.x]: Products with 'Free Shipping' are being factored into shipping rule 'Total Order Quantity' ranges

### DISCUSSION THREAD

**Discussion by icreatestuff** [craftcms/commerce#3122] [General]
### Description
Our store sells a mix of products that have either no shipping requirement, the shipping costs are already included in the product price, and products for which shipping needs calculating and adding in addition. 

When setting a product with the 'Free Shipping' option our understanding is that it would not then be included in any shipping calculations being made on the remainder of the order items.

For example, we have a General shipping method with 3 rules:
1. **1-5 items** - Has 'Total Order Quantity' conditions of between 1 and 5 and a base rate set at £3
2. **6-10 items** - Has 'Total Order Quantity' conditions of between 6 and 10 (no limit) and a base rate set at £5
3. **11+ items** - Has 'Total Order Quantity' conditions of between 11 and 0 (no limit) and a base rate set at £8

An order that contains:
- 1 x Product with 'Free Shipping'
- 5 x Products without 'Free Shipping'

### Expected behavior
Our expectation is that because 1 of those products is set with Free Shipping that it wouldn't count towards the order total in the shipping rule conditions, so the matching rule would be the first (1-5 items).

### Actual behavior
The 2nd rule above is the match that is actually found because there are 6 items in the order.

### Craft CMS version

4.4.1

### Craft Commerce version

4.2.5.1

### PHP version

8.1.11

**Comment by pdaleramirez** [comment]
@icreatestuff I don’t see any bug in the behavior. The rule condition will always see the order quantity of the cart first. I am not sure how you set up the condition to have a single item as free shipping and other items on the different shipping method because base rate is added cost and not a condition.

**Comment by icreatestuff** [comment]
@pdaleramirez I guess my point is that some (or all) of the shipping rules _shouldn't_ match because the 'Total Order Quantity' shouldn't be including those items marked with 'Free Shipping'. Free Shipping items should be completely removed from any conditional rules. 

What's the rationale for having products with 'Free Shipping' still be included in the conditional rules for a shipping method?

**Comment by pdaleramirez** [comment]
@icreatestuff For now the quantity condition rules out free shipping items. We’ll take note of this to add a setting in future versions.
